airbyte_cdk/sources/declarative/incremental/per_partition_with_global.py

```diff
@@ -194,9 +194,11 @@ def get_request_body_json(
         )
 
     def should_be_synced(self, record: Record) -> bool:
-        return self._global_cursor.should_be_synced(
-            record
-        ) or self._per_partition_cursor.should_be_synced(record)
+        return (
```

Comment:
Why change that to use the active cursor only? If we really need to use the active cursor, we should use `_get_active_cursor`

Reply:
```python
self._global_cursor.should_be_synced(
            record
        ) or self._per_partition_cursor.should_be_synced(record)
```

we might get an exception if `self._global_cursor...` returns `False`, we try to get it from `_per_partition_cursor` which cause an exception.

thanks! ref to use `_get_active_cursor`

Reply:
What is the exception that is caused when calling with the per_partition_cursor? I would expect that both work fine

Reply:
Up on this. This is the only remaining question and we can close this after that!

Reply:
ok, so:
https://github.com/airbytehq/airbyte-python-cdk/blob/7ab7e06df0a23ad5262cc73716f3413f498ff898/unit_tests/sources/declarative/extractors/test_record_filter.py#L360-L364 (last 2 tests) failed due to:

``` python3
../airbyte_cdk/sources/declarative/incremental/per_partition_cursor.py:306: in should_be_synced
    return self._get_cursor(record).should_be_synced(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PerPartitionCursor(), record = {'id': 1, 'created_at': '2020-01-03'}

    def _get_cursor(self, record: Record) -> DeclarativeCursor:
        if not record.associated_slice:
            raise ValueError(
                "Invalid state as stream slices that are emitted should refer to an existing cursor"
            )
        partition_key = self._to_partition_key(record.associated_slice.partition)
        if partition_key not in self._cursor_per_partition:
>           raise ValueError(
                "Invalid state as stream slices that are emitted should refer to an existing cursor"
            )
E           ValueError: Invalid state as stream slices that are emitted should refer to an existing cursor
```

we should use active cursor only (do not check per-partition if global cursor returned `False`)

Reply:
Oh, that makes sense! It feels harsh to kill a sync because of this though. Should `should_be_synced` just return `True` if the partition does not exist? I'm curious what @tolik0 thinks about this one